jxl/src/util/shift_right_ceil.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+use std::ops::{Add, Shl, Shr, Sub};
+
+pub trait ShiftRightCeil: Copy {
+    fn shrc<T: Copy>(self, rhs: T) -> Self
+    where
+        Self: Shr<T, Output = Self> + Shl<T, Output = Self>;
+}
+
+impl<S: Copy + Add<Self, Output = Self> + Sub<Self, Output = Self> + From<u8>> ShiftRightCeil
```

Comment:
```suggestion
impl<S: Copy + Add<Self, Output = Self> + Sub<Self, Output = Self> + From<u8>  + Shl<T, Output = Self> + Shr<T, Output = Self>> ShiftRightCeil
```

Don't we also need shifting here?

Reply:
If it compiles this way, we don't need additional constraints ;)